src/Api/DataProvider/AbstractElasticSearchDataProvider.php

```diff
@@ -180,14 +180,21 @@ protected function logStatistics(string $type, array $identifiers, array $result
     {
         $className = substr(\get_class($this), strrpos(\get_class($this), '\\') + 1);
 
-        $this->statsLogger->info('Cover request/response', [
-            'service' => $className,
-            // @TODO Log clientID when authentication implemented, log 'REST_API' for now to allow stats filtering on REST.
-            'clientID' => 'REST_API',
-            'remoteIP' => $request->getClientIp(),
-            'isType' => $type,
-            'isIdentifiers' => $identifiers,
-            'fileNames' => $this->getImageUrls($results),
-        ]);
+        // Defer logging to the kernel terminate event after response has
+        // been delivered.
+        $this->dispatcher->addListener(
+            KernelEvents::TERMINATE,
+            function (TerminateEvent $event) use ($className, $type, $identifiers, $results, $request) {
+                $this->statsLogger->info('Cover request/response', [
+                    'service' => $className,
+                    // @TODO Log clientID when authentication implemented, log 'REST_API' for now to allow stats filtering on REST.
+                    'clientID' => 'REST_API',
+                    'remoteIP' => $request->getClientIp(),
+                    'isType' => $type,
+                    'isIdentifiers' => $identifiers,
+                    'fileNames' => $this->getImageUrls($results),
```

Comment:
Will this `this` reference work in an event outside the scope of the object will `this` not point to something else?